src/index.js

```diff
@@ -243,6 +246,45 @@ module.exports = (web3, options = {}) => {
      * @return {Promise} A promise that resolves to a raw transaction
      */
     async publishVersion (manager, appId, version, provider, directory, contract, from) {
+      const {
+        targetContract,
+        name,
+        params
+      } = await this.publishVersionIntent(manager, appId, version, provider, directory, contract)
+
+      try {
+        const call = targetContract.methods[name](...params)
+
+        // Return transaction to sign
+        return {
+          to: targetContract.options.address,
+          data: call.encodeABI(),
+          gas: Math.round(await call.estimateGas({ from }) * GAS_FUZZ_FACTOR),
+          gasPrice: web3.utils.toWei('10', 'gwei'),
+          nonce: await web3.eth.getTransactionCount(manager)

```

Comment:
Is the `nonce` necessary here? If we let the signer handle this, would it allow us to publish multiple packages with the same key "in parallel" (e.g. without having to wait for the previous publish tx to be mined)?